### ISSUE TITLE
new build for python 3.11

### ISSUE THREAD

**Issue by AntiSol** [DaveMDS/python-efl#4]
Hi,
I'm unable to install from pip on python3.11 on manjaro.

I found this mailing list post and I think I'm probably having the same problem?
https://sourceforge.net/p/enlightenment/mailman/message/37794291/

I also think this might be the same issue as #3 ?

the error i see looks like:
```
  gcc -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -I/home/antisol/code/venv/include -I/usr/include/python3.11 -c efl/eo/efl.eo.c -o build/temp.linux-x86_64-cpython-311/efl/eo/efl.eo.o -I/usr/include/eo-1 -I/usr/include/eina-1 -I/usr/include/eina-1/eina -pthread -I/usr/include/valgrind -fno-var-tracking-assignments -Wno-misleading-indentation -Wno-deprecated-declarations -Wno-unused-variable -Wno-format-security
      efl/eo/efl.eo.c: In function ‘__Pyx_PyErr_GetTopmostException’:
      efl/eo/efl.eo.c:7417:21: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_type’
       7417 |     while ((exc_info->exc_type == NULL || exc_info->exc_type == Py_None) &&
            |                     ^~
      efl/eo/efl.eo.c:7417:51: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_type’
       7417 |     while ((exc_info->exc_type == NULL || exc_info->exc_type == Py_None) &&
            |                                                   ^~
      efl/eo/efl.eo.c: In function ‘__Pyx__ExceptionSave’:
      efl/eo/efl.eo.c:7431:21: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_type’
       7431 |     *type = exc_info->exc_type;
            |                     ^~
      efl/eo/efl.eo.c:7433:19: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_traceback’
       7433 |     *tb = exc_info->exc_traceback;
            |                   ^~
      efl/eo/efl.eo.c: In function ‘__Pyx__ExceptionReset’:
      efl/eo/efl.eo.c:7447:24: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_type’
       7447 |     tmp_type = exc_info->exc_type;
            |                        ^~
      efl/eo/efl.eo.c:7449:22: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_traceback’
       7449 |     tmp_tb = exc_info->exc_traceback;
            |                      ^~
      efl/eo/efl.eo.c:7450:13: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_type’
       7450 |     exc_info->exc_type = type;
            |             ^~
      efl/eo/efl.eo.c:7452:13: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_traceback’
       7452 |     exc_info->exc_traceback = tb;
            |             ^~
      efl/eo/efl.eo.c: In function ‘__Pyx__GetException’:
      efl/eo/efl.eo.c:7534:28: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_type’
       7534 |         tmp_type = exc_info->exc_type;
            |                            ^~
      efl/eo/efl.eo.c:7536:26: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_traceback’
       7536 |         tmp_tb = exc_info->exc_traceback;
            |                          ^~
      efl/eo/efl.eo.c:7537:17: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_type’
       7537 |         exc_info->exc_type = local_type;
            |                 ^~
      efl/eo/efl.eo.c:7539:17: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_traceback’
       7539 |         exc_info->exc_traceback = local_tb;
            |                 ^~
      efl/eo/efl.eo.c: In function ‘__Pyx_AddTraceback’:
      efl/eo/efl.eo.c:464:62: error: invalid use of incomplete typedef ‘PyFrameObject’ {aka ‘struct _frame’}
        464 |   #define __Pyx_PyFrame_SetLineNumber(frame, lineno)  (frame)->f_lineno = (lineno)
            |                                                              ^~
      efl/eo/efl.eo.c:8885:5: note: in expansion of macro ‘__Pyx_PyFrame_SetLineNumber’
       8885 |     __Pyx_PyFrame_SetLineNumber(py_frame, py_line);
            |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for python-efl
```

thanks for your time!

**Comment by TheTechRobo** [comment]
Try these steps:

1. Clone the repository
2. Create a venv (optional, but highly recommended)
3. Install `cython<3` (aka the last version before Cython 3.0.0 - the new versions of Cython don't appear to be compatible)
4. `python setup.py build`

**Comment by AntiSol** [comment]
Thanks @TheTechRobo ! 

To save others time, the version of cython you want is 0.29.36. These commands did the trick for me:
```
git clone <repo>
cd python-efl
python3 -m venv venv
. venv/bin/activate
pip install Cython==0.29.36
pip install packaging
python setup.by build
```

**Comment by TheTechRobo** [comment]
For the record, you can also do `pip install 'Cython<3'` (make sure it's quoted or your shell will think you're trying to do input redirection).

**Comment by AntiSol** [comment]
aha nice, thanks! :)

**Comment by DaveMDS** [comment]
This should be fixed now in 1.26.1 release
